### PR TITLE
mdbm 4.13.0 (new formula)

### PR DESCRIPTION
**Pull request by torden**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

---
This change is [Reviewable](https://reviewable.io/reviews/homebrew/homebrew-core/46723)

### CODE REVIEW

File: Formula/mdbm.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  depends_on "coreutils"
+  depends_on "cppunit"
+  depends_on "make"
+  depends_on "openssl"
+  depends_on "readline"
+
+  def install
+    ENV.delete "CC"
+    ENV.delete "CXX"
```
Comment (fxcoudert) [inline Formula/mdbm.rb]: No, please don't touch these variables.

File: Formula/mdbm.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    system "make", "-j8"
```
Comment (fxcoudert) [inline Formula/mdbm.rb]: Make parallelisation is handled automatically
```suggestion
    system "make"
```

File: Formula/mdbm.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+
+  test do
+    ts_mdbm = "/tmp/test.mdbm"
```
Comment (fxcoudert) [inline Formula/mdbm.rb]: Don't write in `/tmp`. The `test` block is executed in a temporary directory, so just write files in the current directory. (Absolute path is `testpath`, if needed.)

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/mdbm.rb (hunk 1 above)
Comment (torden) [reply Formula/mdbm.rb]: I understood the built environment and will be fix it , thx

File: Formula/mdbm.rb (hunk 2 above)
Comment (torden) [reply Formula/mdbm.rb]: I understood thx

File: Formula/mdbm.rb (hunk 3 above)
Comment (torden) [reply Formula/mdbm.rb]: I Will do fixing by according to code review

File: Formula/mdbm.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+class Mdbm < Formula
+  desc "Y! MDBM a very fast memory-mapped key/value store"
+  homepage "https://github.com/yahoo/mdbm/"
+  url "https://github.com/yahoo/mdbm/archive/v4.13.0.tar.gz"
+  sha256 "99cec32e02639048f96abf4475eb3f97fc669541560cd030992bab155f0cb7f8"
+
+  depends_on "coreutils" => :build
+  depends_on "cppunit" => :build
+  depends_on "make" => :build
+  depends_on "openssl"
```
Comment (fxcoudert) [inline Formula/mdbm.rb]: We now use OpenSSL 1.1:

```suggestion
  depends_on "openssl@1.1"
```

File: Formula/mdbm.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class Mdbm < Formula
+  desc "Y! MDBM a very fast memory-mapped key/value store"
+  homepage "https://github.com/yahoo/mdbm/"
+  url "https://github.com/yahoo/mdbm/archive/v4.13.0.tar.gz"
+  sha256 "99cec32e02639048f96abf4475eb3f97fc669541560cd030992bab155f0cb7f8"
+
+  depends_on "coreutils" => :build
+  depends_on "cppunit" => :build
+  depends_on "make" => :build
```
Comment (fxcoudert) [inline Formula/mdbm.rb]: Why are `make` and `coreutils` required? Aren't the macOS system tools sufficient here?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/mdbm.rb (hunk 4 above)
Comment (torden) [reply Formula/mdbm.rb]: added it, thx

**Comment by fxcoudert** [comment]
```
  * Non-libraries were installed to "/usr/local/opt/mdbm/lib"
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /usr/local/opt/mdbm/lib/libmdbm.dylib.4
```

File: Formula/mdbm.rb (hunk 5 above)
Comment (torden) [reply Formula/mdbm.rb]: removed `make` and mdbm use ginstall in coreutils. (I will try it as soon)

**Comment by iMichka** [comment]
Can all theses changes to the Makefile be upstreamed?
We will need to maintain all these changes if there is a new version one day ... and if the Makefile code is changed, we will need to track down the bugs.

Also, the upstream project does not seem to have had much activity since November 2018. This worries me a little bit, because it looks a little bit dead. But maybe I am wrong, some projects just never need any changes as they are quite simple

**Comment by fxcoudert** [comment]
Acceptance criteria for new formulas state (https://docs.brew.sh/Acceptable-Formulae): “does not need patching to build”. Here there is a lot of makefile hacking and editing the libraries, which is not maintainable in the future. Please get this fixed upstream, and then we can consider it.

In the meantime, you may want to maintain a personal tap: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Pull request closed without merging**